diff --git a/CMake/AbseilDll.cmake b/CMake/AbseilDll.cmake
index 1b89569..2b185a1 100644
--- a/CMake/AbseilDll.cmake
+++ b/CMake/AbseilDll.cmake
@@ -820,7 +820,7 @@ function(absl_make_dll)
       Threads::Threads
       # TODO(#1495): Use $<LINK_LIBRARY:FRAMEWORK,CoreFoundation> once our
       # minimum CMake version >= 3.24
-      $<$<PLATFORM_ID:Darwin>:-Wl,-framework,CoreFoundation>
+      $<LINK_LIBRARY:FRAMEWORK,CoreFoundation>
     )
     set(_dll_compile_definitions "")
     set(_dll_includes "")
diff --git a/absl/time/CMakeLists.txt b/absl/time/CMakeLists.txt
index fe625f2..d37f790 100644
--- a/absl/time/CMakeLists.txt
+++ b/absl/time/CMakeLists.txt
@@ -83,7 +83,7 @@ absl_cc_library(
     Threads::Threads
     # TODO(#1495): Use $<LINK_LIBRARY:FRAMEWORK,CoreFoundation> once our
     # minimum CMake version >= 3.24
-    $<$<PLATFORM_ID:Darwin,iOS,tvOS,visionOS,watchOS>:-Wl,-framework,CoreFoundation>
+    $<LINK_LIBRARY:FRAMEWORK,CoreFoundation>
 )
 
 # Internal-only target, do not depend on directly.
